### PR TITLE
[gn,skia,crashpad,chromium-base] Update to latest

### LINKED ISSUE #17758: [Skia] Bump Version
**Is your feature request related to a problem? Please describe.**
The current port of Skia uses a version of source code from ~~2017~~ EDIT: 2020. The old version has different API and missing functionality compared to the latest version (EDIT: Such as `GrDirectContext` and `SkSamplingOptions`).

**Proposed solution**
Add new versions of Skia to the library. Since vcpkg already has the mechanism for dealing with package versions this should be relatively easy.

**Describe alternatives you've considered**
Use a new package name (e.g., skia-m91). This is more of a last resort.

### PR DESCRIPTION
**Pull request by vejmartin**
Update `gn`, `skia`, and `crashpad` to latest.

- #### What does your PR fix?  
  Fixes #17758 #12123 #18466

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  N/A, Yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Comment by JackBoosY** [comment]
Please get failure logs [here](https://dev.azure.com/vcpkg/public/_build/results?buildId=54305&view=artifacts&pathAsName=false&type=publishedArtifacts).

**Review by JackBoosY — APPROVED** [review]

File: scripts/cmake/vcpkg_install_gn.cmake [hunk 1]
```diff
@@ -25,6 +25,30 @@ Note: includes must be handled separately
 [`vcpkg_build_ninja()`]: vcpkg_build_ninja.md
 #]===]
 
+# Creates a new empty library that depends on the given target for each pair, so that GN can
+#   assign it the correct definitions, which can then be queried via the desc command.
+# Each target pair is the new name, suffixed by the target name (always starts with //)
+#   e.g. newName//full/target:label (new name = newName, target = //full/target:label)
+#   the new target then becomes //extract_public_config:newName
+function(z_vcpkg_install_gn_create_extract_public_config_targets SOURCE_PATH)
```
Comment (strega-nil-ms) [inline scripts/cmake/vcpkg_install_gn.cmake:33]:
```suggestion
function(z_vcpkg_install_gn_create_extract_public_config_targets)
    cmake_parse_arguments(PARSE_ARGV 1 "arg" "" "SOURCE_PATH" "TARGETS")
```

File: scripts/cmake/vcpkg_install_gn.cmake [hunk 2]
```diff
@@ -25,6 +25,30 @@ Note: includes must be handled separately
...
+    file(READ "${SOURCE_PATH}/BUILD.gn" root_build_gn_contents)
```
Comment (strega-nil-ms) [inline scripts/cmake/vcpkg_install_gn.cmake:34]:
```suggestion
    file(READ "${arg_SOURCE_PATH}/BUILD.gn" root_build_gn_contents)
```

File: scripts/cmake/vcpkg_install_gn.cmake [hunk 3]
```diff
@@ -25,6 +25,30 @@ Note: includes must be handled separately
...
+    if(NOT root_build_gn_contents MATCHES "extract_public_config")
+        file(APPEND "${SOURCE_PATH}/BUILD.gn"
```
Comment (strega-nil-ms) [inline scripts/cmake/vcpkg_install_gn.cmake:36]:
```suggestion
        file(APPEND "${arg_SOURCE_PATH}/BUILD.gn" 
```

File: scripts/cmake/vcpkg_install_gn.cmake [hunk 4]
```diff
@@ -25,6 +25,30 @@ Note: includes must be handled separately
...
+#   e.g. newName//full/target:label (new name = newName, target = //full/target:label)
+#   the new target then becomes //extract_public_config:newName
+function(z_vcpkg_install_gn_create_extract_public_config_targets SOURCE_PATH)
+    file(READ "${SOURCE_PATH}/BUILD.gn" root_build_gn_contents)
+    if(NOT root_build_gn_contents MATCHES "extract_public_config")
+        file(APPEND "${SOURCE_PATH}/BUILD.gn" 
+            "\ngroup(\"extract_public_config\"){deps=[\"//extract_public_config\"]}")
+    endif()
+
+    set(targets ${ARGN})
```
Comment (strega-nil-ms) [inline scripts/cmake/vcpkg_install_gn.cmake:40]:
```suggestion
```

File: scripts/cmake/vcpkg_install_gn.cmake [hunk 5]
```diff
@@ -25,6 +25,30 @@ Note: includes must be handled separately
...
+    # create a shared_library entrance for each target pair
+    list(TRANSFORM targets REPLACE "^([^/]+)(.+)$"
+        "shared_library(\"\\1\"){deps=[\"\\2\"]}")
+    # //extract_public_config above is the same as //extract_public_config:extract_public_config
+    list(APPEND targets "group(\"extract_public_config\"){}")
+    string(JOIN "\n" extract_public_config_contents ${targets})
+    file(MAKE_DIRECTORY "${SOURCE_PATH}/extract_public_config")
+    file(WRITE "${SOURCE_PATH}/extract_public_config/BUILD.gn" 
+        "${extract_public_config_contents}")
```
Comment (strega-nil-ms) [inline scripts/cmake/vcpkg_install_gn.cmake:49]:
```suggestion
    list(TRANSFORM arg_TARGETS REPLACE "^([^/]+)(.+)$"
        "shared_library(\"\\1\"){deps=[\"\\2\"]}")
    # //extract_public_config above is the same as //extract_public_config:extract_public_config
    list(APPEND arg_TARGETS "group(\"extract_public_config\"){}")
    list(JOIN arg_TARGETS "\n" extract_public_config_contents)
    file(MAKE_DIRECTORY "${arg_SOURCE_PATH}/extract_public_config")
    file(WRITE "${arg_SOURCE_PATH}/extract_public_config/BUILD.gn" 
        "${extract_public_config_contents}")
```

**Review by strega-nil-ms — CHANGES_REQUESTED** [review]
Making sure to follow the [new guidelines](https://github.com/microsoft/vcpkg/pull/18319)

**Comment by JackBoosY** [comment]
The gl3w regression has nothing to do with this PR.

File: ports/chromium-base/vcpkg.json [hunk 6]
```diff
@@ -0,0 +1,8 @@
+{
+  "name": "chromium-base",
+  "version-string": "86.0.4199.1",
```
Comment (JonLiu1993) [inline ports/chromium-base/vcpkg.json:3]:
```suggestion
  "version-semver": "86.0.4199.1",
```

File: ports/crashpad/vcpkg.json [hunk 7]
```diff
@@ -0,0 +1,13 @@
+{
+  "name": "crashpad",
+  "version-string": "2021-05-30",
```
Comment (JonLiu1993) [inline ports/crashpad/vcpkg.json:3]:
```suggestion
  "version-date": "2021-05-30",
```

File: ports/skia/vcpkg.json [hunk 8]
```diff
@@ -0,0 +1,34 @@
+{
+  "name": "skia",
+  "version-string": "2021-05-30",
```
Comment (JonLiu1993) [inline ports/skia/vcpkg.json:3]:
```suggestion
  "version-date": "2021-05-30",
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by JonLiu1993** [comment]
@vejmartin ,Have you tested the feature metal in port skia?

**Comment by JonLiu1993** [comment]
@vejmartin ,Please take a look:
https://dev.azure.com/vcpkg/public/_build/results?buildId=55984&view=artifacts&pathAsName=false&type=publishedArtifacts

**Comment by PhoebeHui** [comment]
@vejmartin, could you address the failures in CI testing? please let us know if you need the failure logs in CI testing.

**Comment by vejmartin** [comment]
@JonLiu1993 , @PhoebeHui , thank you for your patience. I've re-run x-add-version. There is an issue on macOS where the build for `chromium-base` fails spuriously with errors like:
```
ninja: Entering directory `/Users/vagrant/Data/buildtrees/chromium-base/x64-osx-dbg'
[redacted]
FAILED: obj/base/base/user_metrics.o 
../../../../../../usr/bin/clang++ -MMD -MF obj/base/base/user_metrics.o.d -DSYSTEM_NATIVE_UTF8 -D_LIBCPP_HAS_NO_ALIGNED_ALLOCATION -DCR_XCODE_VERSION= -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__ASSERT_MACROS_DEFINE_VERSIONS_WITHOUT_UNDERSCORES=0 -D_DEBUG -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DBASE_IMPLEMENTATION -I../src/25ce732 -Igen -fno-strict-aliasing -fstack-protector-strong -fcolor-diagnostics -fmerge-all-constants -fcrash-diagnostics-dir=../src/25ce732/tools/clang/crashreports -Xclang -mllvm -Xclang -instcombine-lower-dbg-declare=0 -arch x86_64 -Wno-builtin-macro-redefined -D__DATE__= -D__TIME__= -D__TIMESTAMP__= -no-canonical-prefixes -Wall -Wextra -Wimplicit-fallthrough -Wunreachable-code -Wthread-safety -Wextra-semi -Wunguarded-availability -Wno-missing-field-initializers -Wno-unused-parameter -Wno-c++11-narrowing -Wno-unneeded-internal-declaration -Wno-undefined-var-template -Wno-ignored-pragma-optimize -Wno-implicit-int-float-conversion -Wno-final-dtor-non-final-class -Wno-builtin-assume-aligned-alignment -Wno-deprecated-copy -Wno-non-c-typedef-for-linkage -Wmax-tokens -O0 -fno-omit-frame-pointer -fno-standalone-debug -gdwarf-4 -g2 -isysroot ../../../../../../Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -mmacosx-version-min=10.10.0 -ftrivial-auto-var-init=pattern -fvisibility=hidden -Wheader-hygiene -Wstring-conversion -Wtautological-overlap-compare -Wno-unknown-warning-option -Wno-unknown-pragmas -Wglobal-constructors -Wexit-time-destructors -Wshadow -Wexit-time-destructors -Wno-undefined-bool-conversion -Wno-tautological-undefined-compare -std=c++14 -fno-trigraphs -Wno-trigraphs -stdlib=libc++ -fno-exceptions -fno-rtti -fvisibility-inlines-hidden -include obj/base/base/precompile.h-cc -c ../src/25ce732/base/metrics/user_metrics.cc -o obj/base/base/user_metrics.o
/bin/sh: ../../../../../../usr/bin/clang++: No such file or directory
```
I've seen this happen locally as well and the file `../../../../../../usr/bin/clang++` (i.e. `/usr/bin/clang++`) does exist. Any ideas what the cause might be?

**Comment by PhoebeHui** [comment]
@vejmartin, could you merge changes from master and see if the failure persist?

**Comment by vejmartin** [comment]
@PhoebeHui, I won't be able to look into this PR soon. If anyone would like to continue on it, please feel free to do so.

**Comment by JonLiu1993** [comment]
> @PhoebeHui, I won't be able to look into this PR soon. If anyone would like to continue on it, please feel free to do so.

If you will not be able to view this PR soon, we temporarily convert this PR to a draft. If you continue to process this pr please reopen it

**Comment by JonLiu1993** [comment]
@vejmartin ,Is this PR being worked on?

**Comment by vejmartin** [comment]
@JonLiu1993, I'm not able to work on this PR currently. Please feel free to close it.

**Comment by JonLiu1993** [comment]
@vejmartin, Do you have time to re-launch this PR? I see that you have done a lot of work on this PR before, I hope you can continue, thank you.

**Pull request closed without merging**